Formula/k9s.rb

```diff
@@ -0,0 +1,20 @@
+class K9s < Formula
+  desc "Kubernetes CLI To Manage Your Clusters In Style!"
+  homepage "https://k9scli.io/"
+  url "https://github.com/derailed/k9s.git",
+      :tag      => "v0.21.2",
+      :revision => "977791627860a0febb3c217a5322702da109ecbb"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-ldflags",
+             "-s -w -X github.com/derailed/k9s/cmd.version=#{version}
+             -X github.com/derailed/k9s/cmd.commit=#{stable.specs[:revision]}",
+             *std_go_args
+  end
+
+  test do
+    system "k9s", "version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hey @SMillerDev providing a meaningful assertion might be really difficult as `k9s` only has 3 commands. The closest one to making any sense is just launching the interactive `k9s` without a command and checking for the the text `Boom!! Invalid kubeconfig context detected`. However that test will fail on a regular users machine where a Kubernetes config file exists. In that case the interactive CLI would be launched.

![image](https://user-images.githubusercontent.com/7140800/86511487-0fa27480-bdfa-11ea-8bc8-df368fcaf9a6.png)

Reply:
If it needs to be in `$HOME/.k9s/config.yml` you can set `ENV['HOME']` to `testpath` since that will be an empty directory. That way it'll always fail.

Reply:
@SMillerDev I added a test for the last line as well as a thorough explanation on the expected Kubernetes configuration. k9s needs a Kubernetes config, not its own one to run. Most tools around Kubernetes use either `~/.kube/config` or the `KUBECONFIG` environment variable.
If at some point there is a chance to create a better test I will do so.